[MRG + 1] Add best way ask questions; Fixes #6815

NickleDave opened the pull request:
I added text with links as described in issue #6815.
The formatting checks out on build, see pngs below. In the faq I tried to be helpful and used 'we' in keeping with the tone of the rest of the document.
![screenshot from 2016-07-17 15-49-15](https://cloud.githubusercontent.com/assets/11934090/16903030/87b677ca-4c37-11e6-9b5a-a556ca24faf4.png)
![screenshot from 2016-07-17 15-49-29](https://cloud.githubusercontent.com/assets/11934090/16903031/894a70aa-4c37-11e6-80ec-2d4c43f78e58.png)

Fixes #6815.

NickleDave committed:
add directions for contact to AUTHORS.rst

NickleDave committed:
add 'best way to ask questions' to faq

NickleDave committed:
fix link typos I made in AUTHORS.rst and faq.rst

NickleDave committed:
fix my rst typos again

NickleDave committed:
fix more of my rst typos

nelson-liu commented on AUTHORS.rst:
```diff
@@ -63,4 +63,15 @@ The following people have been core contributors to scikit-learn's development a
   * Nelle Varoquaux
   * `Gael Varoquaux <http://gael-varoquaux.info/blog/>`_
-  * Ron Weiss
+  * Ron Weissz
```
this looks like an unintended change?

nelson-liu commented on AUTHORS.rst:
```diff
@@ -63,4 +63,15 @@ The following people have been core contributors to scikit-learn's development a
+`issue tracker on Github <https://github.com/scikit-learn/scikit-learn/issues>`_
+or the `mailing list <https://mail.python.org/mailman/listinfo/scikit-learn>`_.
+If you are not sure whether you have a found a bug,
+or you have a more general question about
```
this phrase sounds a bit odd...shouldn't it be "if you are not sure whether you have found a bug, or if you have a more general question". could be just me.

nelson-liu commented on AUTHORS.rst:
I'd prefer the wording "convenient search, and access to a community of users that may not be on the mailing list"

NickleDave committed:
fix other typo

nelson-liu commented on doc/faq.rst:
is the scikit-learn list even an appropriate venue for questions about general machine learning...?

NickleDave committed:
Change wording in paragraph after author list

nelson-liu commented on doc/faq.rst:
```diff
@@ -26,6 +26,20 @@ See :ref:`contributing`. Before wanting to add a new algorithm, which is
 
+What's the best way to ask questions about scikit-learn?
+--------------------------------------------------------------
+We of course want to help scikit-learn users. But for broader questions about
```
I think this sentence should be merged with the next. "of course want to" sounds a bit odd to me, how is "Helping users is important to us, but for broader questions about..."

nelson-liu commented on doc/faq.rst:
same comment as above (community wording)

NickleDave committed:
Update faq.rst

NickleDave committed:
Change phrasing

in 'What's the best way to ask questions' section

NickleDave commented:
@nelson-liu made changes based on your comments, thanks

lesteve commented:
@NickleDave just a FYI. It is a lot better to use fix #issueNumber in the PR description rather than in the PR title. This way the issue gets closed automatically when the PR is merged which reduces the maintainers burden.

NickleDave commented:
@lesteve I think I do have it in the first sentence of the description: "I added text with links as described in issue #6815." Is it better to put it at the end as you did?
Also if you have any opinions about the wording I added to the FAQ/Authors pages, I can make more changes.

lesteve commented:
> @lesteve I think I do have it in the first sentence of the description: "I added text with links as described in issue #6815."

Just referencing the issue creates a link but the issue won't get automatically closed when the PR is merged which is very handy. You need to use "fix #issueNumber" to do that. See https://github.com/blog/1506-closing-issues-via-pull-requests for more details.

NickleDave commented:
Ah, got it now. Thanks for clarifying

amueller commented on doc/faq.rst:
```diff
@@ -26,6 +26,20 @@ See :ref:`contributing`. Before wanting to add a new algorithm, which is
+`issue tracker on Github <https://github.com/scikit-learn/scikit-learn/issues>`_
+or the `mailing list <https://mail.python.org/mailman/listinfo/scikit-learn>`_.
+
+Helping users is important to us, but we suggest searching 
```
I would rather phrase it as "The quickest way to get replies to your questions is posting on SO or Cross-validated with the scikit-learn or machine learning tags. These sites may already have an answer for your question and provide a convenient search interface. You can also post questions on the mailing list, though that often has longer turn-around times. Please don't ask general machine learning questions on the mailing list.

Please don't email the authors..."

If you like, you can also mention gitter. But we can also do that in another PR.

NickleDave committed:
Update "best way to ask" with Andreas' phrasing

NickleDave committed:
add Gitter link

nelson-liu commented on doc/faq.rst:
minor nitpick, but stackexchange doesn't allow spaces in tag names. Thus the common tag for ML is "machine-learning"

nelson-liu commented on doc/faq.rst:
i think you wanted backticks here, not single quotes. also this might look better with the S in "Scikit" lowercased

NickleDave committed:
fix typos

nelson-liu commented:
hmm not too sure what's causing the circleci failure....

lesteve commented on AUTHORS.rst:
```diff
@@ -64,3 +64,14 @@ The following people have been core contributors to scikit-learn's development a
   * Ron Weiss
+
+Please do not email the authors directly with bug reports or feature requests.
+To address those concerns, please make use of the
```
To avoid duplicated information, I would just add a link to the "What's the best way to ask questions about scikit-learn" section in the FAQ.

amueller replied:
I'd say "please do not ask authors directly ... and then link"

amueller commented:
the circleci failure is because of all the errors in the doc build that have accumulated since it was last cleaned up. not really related.
LGTM apart from minor comment.

NickleDave committed:
Replace redundant text with link to same in FAQ

amueller commented:
awesome thanks!

nelson-liu commented:
this LGTM as well, thanks @NickleDave

amueller commented:
@GaelVaroquaux @jnothman @NelleV should we remove gitter from this according to the discussion in #6390? I'm fine with removing that but I think the rest of this is quite important.

jnothman commented:
I've not looked at this PR in general, but my opinion is that something
like Gitter may be mentioned as a place where scikit-learn fans, users and
even devs may hang out, but that there are more reliable channels for
raising issues and having questions answered (the primary one being a web
search with good keywords!).

On 25 August 2016 at 05:27, Andreas Mueller <notifications@github.com>
wrote:

> @GaelVaroquaux <https://github.com/GaelVaroquaux> @jnothman
> <https://github.com/jnothman> @NelleV <https://github.com/NelleV> should
> we remove gitter from this according to the discussion in #6390
> <https://github.com/scikit-learn/scikit-learn/issues/6390>? I'm fine with
> removing that but I think the rest of this is quite important.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7032#issuecomment-242180686>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz64ZqSNA1p7c2hi3L0O_JeTsqSlGXks5qjJs6gaJpZM4JOUAd>
> .
>

amueller commented:
fixed in #7309
